src/test/kotlin/blackjack/model/card/CardDeckTest.kt

```diff
@@ -0,0 +1,99 @@
+package blackjack.model.card
+
+import blackjack.model.strategy.NormalStrategy
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class CardDeckTest {
+    @Test
+    fun `카드 덱에서 카드 한장을 받는다`() {
+        val cardDeck = CardDeck(NormalStrategy().shuffle())
+        val result = cardDeck.get()
+
+        result shouldBe Card(CardNumber.ACE, CardSuit.클럽)
+    }
+
+    @Test
+    fun `카드 덱에서 카드 한장을 받으면 덱의 카드 수가 1 감소한다`() {
+        val cardDeck = CardDeck(NormalStrategy().shuffle())
+        val beforeSize = cardDeck.deck.size
+        cardDeck.get()
+        val afterSize = cardDeck.deck.size
+
+        beforeSize - 1 shouldBe afterSize
+    }
+
+    @Test
+    fun `카드 덱에 카드를 추가하면 덱의 카드 수가 1 증가한다`() {
+        val cardDeck = CardDeck()
+        val beforeSize = cardDeck.deck.size
+        cardDeck.add(Card(CardNumber.ACE, CardSuit.클럽))
+        val afterSize = cardDeck.deck.size
+
+        beforeSize + 1 shouldBe afterSize
+    }
+
+    @Test
+    fun `카드 덱에 카드가 없는 경우 카드를 받을 수 없다`() {
+        val cardDeck = CardDeck()
+
+        assertThrows<IllegalStateException> {
+            cardDeck.get()
+        }
+    }
+
+    @Test
+    fun `카드 덱에 이미 존재하는 카드를 받을 수 없다`() {
+        val cardDeck = CardDeck(NormalStrategy().shuffle())
```

Comment:
테스트 setup의 범위를 줄여주는건 어떨까요?

실제로 exception의 원인이 되는 조건을 명확하게 주어주는 것이 더 좋을 것 같습니다.

```suggestion
        val cardDeck = CardDeck(listOf(
            Card(CardNumber.ACE, CardSuit.클럽)
        ))
```

Reply:
실제 게임의 규칙을 생각해보면 모든 카드가 있는 경우에 카드를 추가한다는게 어색하긴 하네요 ㅎㅎ;
반영해 두겠습니다!